api/v1/istio_types.go

```diff
@@ -0,0 +1,186 @@
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	// Resources specifies the compute resources required for the deployment.
+	// +optional
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+	// Tolerations specifies the tolerations for the deployment.
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+// IstioCNIDaemonsetSpecTemplate defines the template for customizing the Istio CNI Daemonset.
+type IstioCNIDaemonsetSpecTemplate struct {
+	// Spec allows users to specify custom fields for the Istio CNI Daemonset.
+	// +optional
+	Spec *IstioCNIDaemonsetPodSpec `json:"spec,omitempty"`
+}
+
+// IstioCNIDaemonsetSpec defines the spec for customizing the Istio CNI Daemonset.
+type IstioCNIDaemonsetSpec struct {
+	// Template allows users to specify custom fields for the Istio CNI Daemonset.
+	// +optional
+	Template *IstioCNIDaemonsetSpecTemplate `json:"template,omitempty"`
+}
+
+// IstioCNIDaemonset defines customized settings for the Istio CNI plugin.
+type IstioCNIDaemonset struct {
+	// Spec allows users to specify custom fields for the Istio CNI Daemonset.
+	// +optional
+	Spec *IstioCNIDaemonsetSpec `json:"spec,omitempty"`
+}
+
+// ZTunnelDaemonsetPodSpec defines the pod spec for customizing the ZTunnel Daemonset.
+type ZTunnelDaemonsetPodSpec struct {
+	// Affinity specifies the affinity for the deployment.
+	// +optional
+	Affinity *corev1.Affinity `json:"affinity,omitempty"`
+	// NodeSelector specifies the node affinity for the deployment.
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	// Resources specifies the compute resources required for the deployment.
+	// +optional
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+	// Tolerations specifies the tolerations for the deployment.
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+// ZTunnelDaemonsetSpecTemplate defines the template for customizing the ZTunnel Daemonset.
+type ZTunnelDaemonsetSpecTemplate struct {
+	// Spec allows users to specify custom fields for the ZTunnel Daemonset.
+	// +optional
+	Spec *ZTunnelDaemonsetPodSpec `json:"spec,omitempty"`
+}
+
+// ZTunnelDaemonsetSpec defines the spec for customizing the ZTunnel Daemonset.
+type ZTunnelDaemonsetSpec struct {
+	// Template allows users to specify custom fields for the ZTunnel Daemonset.
+	// +optional
+	Template *ZTunnelDaemonsetSpecTemplate `json:"template,omitempty"`
+}
+
+// ZTunnelDaemonset defines customized settings for the ZTunnelDaemonset component.
+type ZTunnelDaemonset struct {
+	// Spec allows users to specify custom fields for the ZTunnel Daemonset.
+	// +optional
+	Spec *ZTunnelDaemonsetSpec `json:"spec,omitempty"`
+}
+
+// IstioSpec defines the desired state of Istio
+type IstioSpec struct {
+	// Istiod defines the resource requirements and node selector for the Istio deployment.
+	// +optional
+	Istiod *IstiodDeployment `json:"istiod,omitempty"`
```

Comment:
Sorry I missed these in the design, but for consistency these should all be suffixed with the type of object they are.

See the Installation spec as an example: https://github.com/tigera/operator/blob/cdb970b7f2a9d05f388d33a4bb3241d6f76f0c0f/api/v1/installation_types.go#L163-L180